### PR TITLE
[Fix] Torch random.shuffle(axis=-1) mishandles negative axes

### LINKED ISSUE #23118: [Bug] Torch `random.shuffle(axis=-1)` mishandles negative axes
### Description

`keras.random.shuffle(..., axis=-1)` produces incorrect results with the Torch backend.

Negative axes are not canonicalized before building shuffle indices, so `axis=-1` is treated as the literal axis value instead of the last dimension. 

### Minimal reproduction

```python
import os
os.environ["KERAS_BACKEND"] = "torch"

import keras
import numpy as np

x = keras.ops.reshape(keras.ops.arange(12), (3, 4))
y = keras.random.shuffle(x, axis=-1, seed=123)

x_np = keras.ops.convert_to_numpy(x)
y_np = keras.ops.convert_to_numpy(y)

print(x_np)
print(y_np)

for before, after in zip(x_np, y_np):
    assert sorted(after.tolist()) == sorted(before.tolist()), (before, after)
```
### Expected behavior

axis=-1 should behave the same as axis=x.ndim - 1

There is need of canonicalization in `src/backend/torch/random.py` in the shuffle implementation.

### PR DESCRIPTION
**Pull request by ChiragSW**
## Description
Fixes: #23118 
axis now goes through canonicalize_axis(). Added a small test for axis=-1.

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

**Comment by ChiragSW** [comment]
I don't know why the one numpy test is failing. It's passing on my machine with numpy backend.
Logs:
```bash
PS C:\Users\watta\hive\keras> $env:KERAS_BACKEND='numpy'   
PS C:\Users\watta\hive\keras> pytest keras/src/ops/image_test.py::ImageOpsDtypeTest::test_perspective_transform_bfloat16
===================================================== test session starts =====================================================
platform win32 -- Python 3.10.9, pytest-9.0.2, pluggy-1.6.0 -- C:\Users\watta\AppData\Local\Programs\Python\Python310\python.exe
cachedir: .pytest_cache
rootdir: C:\Users\watta\hive\keras
configfile: pyproject.toml
plugins: anyio-4.11.0, langsmith-0.4.38
collected 1 item                                                                                                               

keras/src/ops/image_test.py::ImageOpsDtypeTest::test_perspective_transform_bfloat16 PASSED                               [100%]

====================================================== 1 passed in 2.28s ======================================================
```

@hertschuh please help me out here

**Review by hertschuh — APPROVED** [review]
Thanks for the fix
